pymeasure/instruments/keysight/keysightN7776C.py

```diff
@@ -0,0 +1,256 @@
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+import numpy as np
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+WL_RANGE = [1480,1620]
+LOCK_PW = 1234
+
+class KeysightN7776C(Instrument):
+    """
+    This represents the Keysight N7776C Tunable Laser Source interface.
+    .. code-block:: python
+        laser = N7776C( address )
+        laser.sweep_wl_start = 1550
+        laser.sweep_wl_stop = 1560
+        laser.sweep_speed = 1
+        laser.sweep_mode = 'CONT'
+        laser.output = 1
+        while laser.sweep == 1:
+            continue
+        laser.output = 0
+    
+    """
+    def __init__(self, address, **kwargs):
+        super(KeysightN7776C, self).__init__(
+            address, "N7776C Tunable Laser Source",**kwargs)
+
+    def reset(self):
+        """
+        Reset the laser controller to its default state.
+        """
+        self.write('*RST')
+
+    def set_power(self,value,unit='mW'):
+        """
+        Function wrapper to set the power in an arbitrary unit
+        """
+        if not unit in ['dBm','mW']:
+            raise ValueError('Unknown Power Unit.')
+
+        self._output_power_unit = unit
+        if unit == 'mW':
+            self._output_power = value*1e-3 #Conversion to mW
+        else:
+            self._output_power = value
+
+    def get_power(self,unit_output=False):
+        """
+        Function wrapper to get the power including the unit directly if neccessary.
+        """
+        power_reading = self._output_power
+        power_unit = self._output_power_unit
+        if power_unit == 'mW':
+            if unit_output:
+                return (power_reading*1e3,power_unit)
+            else:
+                return power_reading*1e3
+        else:
+            if unit_output:
+                return (power_reading,power_unit)
+            else:
+                return power_reading
+    def lock(self):
+        self._locked = True
+        
+    def unlock(self):
+        self._locked = False
+        
+    def locked(self):
+        return self._locked
+
+    _locked = Instrument.control(':LOCK?',':LOCK %g,'+str(LOCK_PW),
+                                   """ Boolean property controlling the lock state (True/False) of the laser source""",
+                                   validator=strict_discrete_set,
+                                   values=[True,False],
+                                   set_process=lambda v: int(v),
+                                   get_process=lambda v: bool(v))
```

Comment:
A more pymeasure-idiomatic way for these 4 methods would be to just have a `locked = Instrument.control...` as you already have; instead of `instr.lock()` you use `instr.locked = True`.

Also, a smoother way to implement the booleanness is
```
        validator=strict_discrete_set,
        map_values=True,
        values={True: 1, False: 0},
```
(I have not commented again if that appears elsewhere)